File: UMS.Core/Models/StudentSubject.cs
using System;
using System.Collections.Generic;

namespace UMS.Core.Models;

public partial class StudentSubject
{
    public int Id { get; set; }

    public int StudentId { get; set; }

    public int SubjectId { get; set; }

    public decimal? Grade { get; set; }

    public DateTime? CreationDate { get; set; }

    public string? ModificationUserId { get; set; }

    public DateTime? ModificationDate { get; set; }

    public bool IsDeleted { get; set; }

    public string? DeletionUserId { get; set; }

    public DateTime? DeletionDate { get; set; }

    public string CreatorUserId { get; set; } = null!;

    public virtual Student Student { get; set; } = null!;

    public virtual Subject Subject { get; set; } = null!;
}

File: ums.client/src/shared/models/StudentSubject.ts
import { Subject } from "./Subject";

export class StudentSubject {
  id: number;
  studentId: number;
  subjectId: number;
  userId: number;
  grade: number | null | undefined;
  subject: Subject;
  isDeleted: boolean;

  constructor(id: number, studentId: number, subjectId: number, userId: number, grade: number | null | undefined, subject: Subject, isDeleted: boolean) {
    this.id = id;
    this.studentId = studentId;
    this.subjectId = subjectId;
    this.userId = userId;
    this.grade = grade;
    this.subject = subject;
    this.isDeleted = isDeleted;
  }
}
